app/cdap/components/SourceControlManagement/LocalPipelineListView/CommitModal.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import ConfirmationModal from 'components/shared/ConfirmationModal';
+import { LoadingAppLevel } from 'components/shared/LoadingAppLevel';
+import React, { useState } from 'react';
+import T from 'i18n-react';
+import { TextareaAutosize } from '@material-ui/core';
+import Alert from 'components/shared/Alert';
+import { pushSelectedPipelines } from '../store/ActionCreator';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+
+const PREFIX = 'features.SourceControlManagement.push';
+
+interface ICommitModalProps {
+  isOpen: boolean;
+  onToggle: () => void;
+  onSubmit?: (commitMessage: string) => void;
+  loadingMessage?: string;
+  pipelineName?: string;
+}
+
+export const CommitModal = ({
+  isOpen,
+  onToggle,
+  onSubmit,
+  loadingMessage = null,
+  pipelineName,
+}: ICommitModalProps) => {
+  const [commitMessage, setCommitMessage] = useState(null);
+  const [pushError, setPushError] = useState(null);
+  const [pushSuccess, setPushSuccess] = useState(null);
+  const [pushLoadingMessage, setPushLoadingMessage] = useState(null);
+
+  const resetPushErrorAndSuccess = () => {
+    setPushError(null);
+    setPushSuccess(null);
+  };
+
+  const handlePipelinePush = () => {
+    const selectedPipelines = [pipelineName];
+    const payload = {
+      commitMessage,
+    };
+    onToggle();
+    pushSelectedPipelines(
+      getCurrentNamespace(),
+      selectedPipelines,
+      payload,
+      setPushLoadingMessage
+    ).subscribe(
+      (res: any) => {
+        if (res.statusCode !== 200) {
+          setPushError(res.message);
+          setPushSuccess(false);
+        } else {
+          setPushSuccess(true);
+        }
+      },
+      (err) => {
+        // most likely it will never arrive here, since the request error
```

Comment:
why is the err here then?

Reply:
If you discussed this offline, can you put the answer here? The concern is that this block will eat the error and it will never surface.

Reply:
Yes we discussed about it. The underneath request will return either success or failure response to the `next` block, will never enter the `error` block, so removed it.